src/types.ts

```diff
@@ -104,17 +104,44 @@ export type Blockchain =
       ens?: NamingServiceSource;
       zns?: NamingServiceSource;
       cns?: NamingServiceSource;
-      web3Provider?: Web3Provider;
+      web3Provider?: Provider;
+      provider?: Provider;
     };
 
 export type API = {
   url: string;
 };
 
-export interface Web3Provider {
+export interface Provider {
   sendAsync: (method: string, params: any) => Promise<any>;
 }
 
+/**
+ * @deprecated
+ */
+export interface Web3Provider {
+  sendAsync: (method: string, params: any) => Promise<any>; 
+}
```

Comment:
```suggestion
export type Web3Provider = Provider;
```
to eliminate code duplication